hpobench/benchmarks/ml_mmfb/tabular_benchmark.py

```diff
@@ -0,0 +1,230 @@
+
+    def _seeds_used(self) -> List:
+        return self.table.seed.unique().tolist()
+
+    def sample_hyperparamer(self, n: int = 1) -> Union[CS.Configuration, List]:
+        return self.configuration_space.sample_configuration(n)
+
+    def sample_fidelity(self, n: int = 1) -> Union[CS.Configuration, List]:
+        return self.fidelity_space.sample_configuration(n)
+
+    def get_global_min(self, metric: str = "acc"):
+        """ Retrieves the minimum (1 - metric) for train, validation and test splits
+        """
+        assert metric in self.global_minimums.keys(), \
+            "Not a valid metric: {}".format(list(self.global_minimums.keys()))
+        return self.global_minimums[metric]
+
+    def get_max_fidelity(self) -> Dict:
+        max_fidelity = dict()
+        for hp in self.fidelity_space.get_hyperparameters():
+            max_fidelity[hp.name] = np.sort(hp.sequence)[-1]
+        return max_fidelity
+
+    def get_fidelity_range(self) -> List:
+        fidelities = []
+        for hp in self.fidelity_space.get_hyperparameters():
+            if not isinstance(hp, CS.Constant) and len(hp.sequence) > 1:
+                fidelities.append((hp.name, hp.sequence[0], hp.sequence[-1]))
+        return fidelities
+
+    def _search_dataframe(self, row_dict, df):
+        query_stmt = self._build_query(row_dict)
+        result = df.query(query_stmt)
+        # TODO: What happens in this case? The objective function raises a TypeError.
+        if len(result) == 0:
+            return None
+        return result.iloc[0].loc['result']
+
+        # TODO: This created an out-of-bounds error. The idx mask should have been 2d, but was 1d.
+        # # https://stackoverflow.com/a/46165056/8363967
+        # mask = np.array([True] * df.shape[0])
+        # for i, param in enumerate(df.drop("result", axis=1).columns):
+        #     mask *= df[param].values == row_dict[param]
+        # idx = np.where(mask)
+        # if len(idx) != 1:
+        #     return None
+        # idx = idx[0][0]
+        # result = df.iloc[idx]["result"]
+        # return result
+
+    @staticmethod
+    def _build_query(row_dict: Dict) -> str:
+        query = ''
+        for i, (param_name, param_value) in enumerate(row_dict.items()):
+            if i != 0:
+                query += ' & '
+            query += f'{param_name} == {param_value}'
+        return query
+
+    def _objective(
+            self,
+            config: Dict,
+            fidelity: Dict,
+            seed: Union[int, None] = None,
+            metric: Union[str, None] = "acc",
+            evaluation: Union[str, None] = ""
+    ) -> Dict:
+
+        metric_str = ', '.join(list(metrics.keys()))
+        assert metric in list(metrics.keys()), f"metric not found among: {metric_str}"
+        score_key = f"{evaluation}_scores"
+        cost_key = f"{evaluation}_scores"
+
+        key_path = dict()
+        # TODO: Dicts are unordered. This does not have to have an effect.
```

Comment:
not sure what you meant here, i.e., the effect of `np.sort`?

Reply:
I just thought that we can remove the ` np.sort(..)`. 
Because if you go in a increasing order through the hp-names and then insert them into a dictionary that has no ordering, this operation does not change anything. 
Or why are you using the sort function here?

Reply:
hmm not 100% sure since there are a couple of reasons
for a dict, yes I agree, doesn't make sense, or rather is immaterial
however, I think the DataFrame columns are in this order and when I am looking up HP names and creating the query order, it's important to be in that order 
`np.sort` probably is not required since I think the `get_hyperparameter_names` does so internally or that's how dict keys are returned probably
however, since a lot of steps in the pipeline (benchmark collection) depend on it, having `np.sort` was kind of a strict check I had enforced myself to ensure parity